src/popup/components/form/FormAccountInput.vue

```diff
@@ -0,0 +1,246 @@
+        :placeholder="placeholder"
+        :value="modelValue"
+        :rows="1"
+        :disabled="selectedAddress || readonly"
+        @keydown.enter.prevent="handleEnterClick"
+        @input="(payload) => handleInput(payload as InputEvent)"
+        @blur="(payload) => handleInput(payload as unknown as InputEvent)"
+      />
+      <template v-if="selectedAddress">
+        <Truncate
+          :str="selectedAddress.name"
+          class="truncated-text"
+        />
+        <AddressTruncated
+          :address="selectedAddress.address"
+          :protocol="protocol"
+        />
+        <BtnIcon
+          v-if="!hideClearIcon"
+          :icon="CircleCloseIcon"
+          data-cy="clear-address-button"
+          class="close-icon"
+          size="sm"
+          @click="clearAddress"
+        />
+      </template>
+    </template>
+
+    <template
+      v-for="(index, name) in $slots"
+      #[name]
+    >
+      <slot :name="name" />
+    </template>
+  </InputAccount>
+</template>
+
+<script lang="ts">
+import { useAddressBook } from '@/composables';
+import { PROTOCOLS } from '@/constants';
+import CircleCloseIcon from '@/icons/circle-close.svg?vue-component';
+import AddressTruncated from '@/popup/components/AddressTruncated.vue';
+import InputAccount from '@/popup/components/InputAccount.vue';
+import Truncate from '@/popup/components/Truncate.vue';
+import BtnIcon from '@/popup/components/buttons/BtnIcon.vue';
+import { IAddressBookEntry, Protocol } from '@/types';
+import {
+  defineComponent,
+  nextTick,
+  onMounted, PropType,
+  ref,
+  watch,
+} from 'vue';
+
+const SIZES = ['xxs', 'xs', 'sm', 'rg', 'md'];
+
+export default defineComponent({
+  components: {
+    BtnIcon,
+    InputAccount,
+    Truncate,
+    AddressTruncated,
+  },
+  props: {
+    type: { type: String, default: '' },
+    modelValue: { type: String, default: '' },
+    placeholder: { type: String, default: '' },
+    textLimit: { type: Number, default: null },
+    resizable: { type: Boolean, default: true },
+    size: {
+      type: String,
+      default: null,
+      validator: (val: string) => SIZES.includes(val),
+    },
+    options: {
+      type: Array,
+      default() {
+        return [];
+      },
+    },
+    enterSubmit: Boolean,
+    autoHeight: Boolean,
+    readonly: Boolean,
+    hideClearIcon: { type: Boolean, default: false },
+    protocol: { type: String as PropType<Protocol>, default: null },
+  },
+  emits: ['update:modelValue', 'submit'],
+  setup(props, { emit }) {
+    // const addressBook = useStorageRef<IAddressBookEntry[]>([], STORAGE_KEYS.addressBook);
+    const {
+      addressBookFiltered,
+      protocolFilter,
+    } = useAddressBook();
+    protocolFilter.value = props.protocol;
+
+    const selectedAddress = ref<IAddressBookEntry | undefined>();
+
+    const textarea = ref<HTMLTextAreaElement>();
+    const height = ref<string | undefined>();
+    function handleInput(event: InputEvent) {
```

Comment:
I suggest to maintain consistency when using empty lines. It's not a strict rule for us, but in most cases we try to not have empty lines between constant definitions but we add one empty line before function definitions.